code/implementation/docker/machine/src/main/java/io/cattle/platform/docker/machine/api/MachineConfigLinkHandler.java

```diff
@@ -28,6 +32,13 @@ public boolean handles(String type, String id, String link, ApiRequest request)
 
     @Override
     public Object link(String name, Object obj, ApiRequest request) throws IOException {
+        SchemaFactory schemaFactory = request.getSchemaFactory();
+        Schema machineSchema = schemaFactory == null ? null :schemaFactory.getSchema(MACHINE_KIND);
+        boolean canCall = machineSchema != null && machineSchema.getCollectionMethods()
```

Comment:
Duplicated logic, right? Maybe just pull out into a static method on this or the link filter so that you can reuse it.